nemo/utils/callbacks/nemo_model_checkpoint.py

```diff
@@ -257,7 +297,52 @@ def _ema_callback(self, trainer: 'pytorch_lightning.Trainer') -> Optional[EMA]:
                 ema_callback = callback
         return ema_callback
 
+    @staticmethod
+    def format_checkpoint_unfinished_marker_path(checkpoint_filepath: Union[Path, str]) -> Path:
```

Comment:
Needs documentation of when the barrier after before should be used.

Reply:
I extended comments to better explain the purpose of the barriers.